Fix the Docker configuration and add testing for the configuration on pull requests

mcdonnnj opened the pull request:
<!-- GitHub renders PRs such that soft line breaks are treated as hard
line breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  This is the reason for the markdownline
disable directive just below.

For more details see:
https://github.com/github/markup/issues/1050#issuecomment-294654762 -->
<!-- markdownlint-disable MD013 -->
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request fixes the Docker configuration so that you can successfully build a Docker image again. This was done by switching the apt SourceList configuration to using the Debian Archive. It also adds the Debian Backports repository to mirror what's available on CyHy instances as well as adds a GitHub Actions workflow to test that a Docker image can be successfully built for pull requests.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

@dav3r noticed that you could not build a Docker image anymore. This is due to the Debian Buster package repositories being removed from the regular apt repository and only being available from the Debian Archive.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

I verified that I could successfully build a Docker image with this updated configuration.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Use the full identifier for source Docker images

This helps ensure that when a Docker image is built the expacted source
image is used regardless of what repository is configured as the
default on the host system. It also makes our Dockerfiles more
seamlessly convertible to using the GitHub Container Registry or any
other Open Container Initiative (OCI) compatible registry.

mcdonnnj committed:
Use the Debian Archive in the Docker image configuration

Since the Debian Buster apt repository has been archived we must change
the apt repository configuration to use the Debian Archive for
packages.

mcdonnnj committed:
Add the Debian Backports repository to the Docker image

This mirrors the fact that we do this on actual CyHy instances.

mcdonnnj committed:
Add a workflow to test the Docker configuration

This workflow will trigger on pull requests and merge groups to ensure
that the Docker configuration will build an image successfully.

dv4harr10 commented:
Hi Nick @mcdonnnj , Dockerfile line 25 has one misspelling for 'configuration'.

jsf9k committed:
Fix typo

Co-authored-by: dv4harr10

dav3r reviewed (approved):
💪 💼  Good thinking adding in that test workflow.  Please make sure that the PR title ~and description~ mentions it as well.

mcdonnnj commented:
> 💪 💼 Good thinking adding in that test workflow. Please make sure that the PR title ~and description~ mentions it as well.

Title updated.

mcdonnnj committed:
Update tracked issue in a comment

In the new workflow I copied over content from cisagov/ncats-webui that
was similar to the needs in this project. That included a comment block
that references an issue in that repository. I have updated the comment
to refer to a newly-created issue in this repository for better
tracking/visibility.

mcdonnnj commented on .github/workflows/test-docker-configuration.yml:
```diff
@@ -0,0 +1,94 @@
+          # monitor_permissions equal to "true".
+          #
+          # TODO: Re-enable this functionality when practical.  See
+          # cisagov/ncats-webui#52 for more details.
```
Please see [4b54deb](https://github.com/cisagov/cyhy-core/pull/102/commits/4b54debb4ab964876f40351b5f322805a953c65a) for an updated reference to an issue in this project. I chose to add tracking for this repository instead of reusing the cisagov/skeleton-generic issue because this project is not included in our Lineage tree due to it being written in Python 2.
